### PR TITLE
SIMSBIOHUB-585+586 Bug Fixes

### PR DESCRIPTION
**Pull request by MacDeluca**
## Links to Jira Tickets

- [SIMSBIOHUB-585 Cancel Dialog](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-585)
- [SIMSBIOHUB-586 Species Survey Save Bug](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-586)

## Description of Changes

- Fixed bug with confirmation dialog incorrectly rendering when saving project forms
- Fixed bug with survey page not being able to be saved when adding animals
- Created custom hook ```useUnsavedChangesDialog```
- Note: All the forms/pages (maybe 5/6 different places) that were using the confirmation dialog were duplicating very similar code to handle location changes. This seemed like a good place to have a custom hook to handle the confirmation dialog and location change interceptor.

## Testing Notes

- Should be able to save a survey when adding animals
- Confirmation dialogs should render for all forms correctly when trying to navigate to a different page.
- Both actions on the confirmation dialogs should work in all situations correctly

### CODE REVIEW

File: app/src/hooks/useUnsavedChanges.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * Generates the cancel dialog props
+   *
+   * @param {string} pathname - Path to redirect to on confirmation
+   * @returns {IYesNoDialogProps} Cancel dialog props
+   */
+  const getCancelDialogProps = (pathname: string): IYesNoDialogProps => {
+    return {
+      dialogTitle: EditSurveyI18N.cancelTitle,
+      dialogText: EditSurveyI18N.cancelText,
```
Comment (NickPhura) [inline app/src/hooks/useUnsavedChanges.tsx]: We'll need to make this text generic or configurable

Comment (NickPhura) [reply app/src/hooks/useUnsavedChanges.tsx]: Personally I think generic would be fine. One less thing to have to worry about every time it gets used.

Comment (MacDeluca) [reply app/src/hooks/useUnsavedChanges.tsx]: 🌵

**Review by NickPhura — APPROVED** [review]
🐐

**Pull request merged**